### PR TITLE
fix: Miscellaneous bugs

### PR DESCRIPTION
**Pull request by pranavxc**
## Change Summary

- [x] #4580
- [x] #4293
- [x] Update refresh token based token generation url in ui
- [x] #4852
- [x] #2014
- [x] #4255
- [x] #4863

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/components/smartsheet/Grid.vue [hunk 1]
```diff
@@ -201,7 +201,7 @@ const { isCellSelected, activeCell, handleMouseDown, handleMouseOver, handleCell
       const cmdOrCtrl = isMac() ? e.metaKey : e.ctrlKey
       const altOrOptionKey = e.altKey
       if (e.key === ' ') {
-        if (activeCell.row != null && !editEnabled) {
+        if (activeCell.row != null && !editEnabled && hasEditPermission?.value) {
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/Grid.vue:204]: shouldn't it be `hasEditPermission`?

**Review by wingkwong — COMMENTED** [review]
just wanna double confirm that if the API path changes should also apply to `RestAuthCtrl.ts`? or stay as it is for v1?

**Comment by pranavxc** [comment]
> just wanna double confirm that if the API path changes should also apply to `RestAuthCtrl.ts`? or stay as it is for v1?

It's used in v1 and we can ignore it.

**Review by dstala — CHANGES_REQUESTED** [review]
- [ ] Shared view, mark single select field as Read Only
  
Screenshot 2023-01-17 at 3 48 31 PM

**Comment by dstala** [comment]
@pranavxc if we remove some of the options from filter for checkbox, how will this be handled during migration if someone had configured such filters for checkbox?

**Comment by pranavxc** [comment]
> @pranavxc if we remove some of the options from filter for checkbox, how will this be handled during migration if someone had configured such filters for checkbox?

The current filter `comparison_op` will be visible in the dropdown even if not allowed.

File: packages/nc-gui/composables/useViewData.ts [hunk 2]
```diff
@@ -469,6 +478,48 @@ export function useViewData(
...
+      // if first page, do nothing
+      if (currentPage === 1) return message.info(t('msg.info.noMoreRecords'))
+
+      await changePage(currentPage - 1)
+      siblingRowIndex = formattedData.value.length - 1
+
+      // if next row index is greater than total rows in current view
+      // then load next page of formattedData and set next row index to 0
+    } else if (siblingRowIndex >= formattedData.value.length) {
+      console.log(paginationData?.value)
```
Comment (wingkwong) [inline packages/nc-gui/composables/useViewData.ts:503]: can be removed

File: packages/nc-gui/components/smartsheet/toolbar/ColumnFilter.vue [hunk 3]
```diff
@@ -213,24 +219,21 @@ defineExpose({
               dropdown-class-name="nc-dropdown-filter-comp-op"
               @change="filterUpdateCondition(filter, i)"
             >
-              <a-select-option v-for="compOp in comparisonOpList" :key="compOp.value" :value="compOp.value" class="">
-                {{ compOp.text }}
-              </a-select-option>
+              <template v-for="compOp in comparisonOpList" :key="compOp.value">
```
Comment (wingkwong) [inline packages/nc-gui/components/smartsheet/toolbar/ColumnFilter.vue:222]: `compOp of comparisonOpList`

File: packages/nc-gui/lib/enums.ts [hunk 4]
```diff
@@ -98,3 +98,8 @@ export enum DataSourcesSubTab {
   Misc = 'Misc',
   Edit = 'Edit',
 }
+
+export enum RowNavDir {
+  NEXT,
+  PREV,
+}
```
Comment (wingkwong) [inline packages/nc-gui/lib/enums.ts:105]: we have `NavigateDir` already which is same as `RowNavDir`. can we just use that?

Comment (pranavxc) [reply packages/nc-gui/lib/enums.ts:105]: Oh thanks, let me update

**Review by dstala — APPROVED** [review]

**Pull request merged**